### PR TITLE
Document `slowmode_delay` param in StageChannel and VoiceChannel, add VocalGuildChannel.slowmode_delay

### LINKED ISSUE #9110: VoiceChannel Slowmode
### Summary

VoiceChannels can have a slowmode. This is not reflected in Discord.py

### What is the feature request for?

The core library

### The Problem

I want to be able to check the slowmode of a VoiceChannel, the editing part is fine (using the `VoiceChannel.edit` function with a `slowmode_delay` kwarg works as I've tested, but not documented). I have found no documentation for reading this attribute.

### The Ideal Solution

I wish for there to be a `slowmode_delay` attribute for `VoiceChannel`

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by rdrescher909**
## Summary

Documents the ability to use the `slowmode_delay`parameter in StageChannel and VoiceChannel `edit` methods.

Fixes #9110

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Zephyrkul** [comment]
This is only a partial fix for 9110 - while it does allow setting the slowmode delay, there's still no attribute on VoiceChannel for finding out what the current slowmode delay is.

**Comment by rdrescher909** [comment]
Working on that part right now :)

File: discord/channel.py [hunk 1]
```diff
@@ -1395,7 +1398,6 @@ async def edit(self, *, reason: Optional[str] = None, **options: Any) -> Optiona
         rtc_region: Optional[:class:`str`]
             The new region for the voice channel's voice communication.
             A value of ``None`` indicates automatic voice region detection.
-
```
Comment (scrazzz) [inline discord/channel.py:1398]: This newline shouldn't be removed, it's like this everywhere.

**Comment by Zephyrkul** [comment]
As far as I'm aware, StageChannels are not messageable, so adding a slowmode attribute to that could be confusing or misleading

**Comment by rdrescher909** [comment]
> As far as I'm aware, StageChannels are not messageable, so adding a slowmode attribute to that could be confusing or misleading

They do have a slow mode slider in the UI. For the sake of completeness I figured it should be included.

**Comment by Zephyrkul** [comment]
> They do have a slow mode slider in the UI. For the sake of completeness I figured it should be included.

That is very odd. Given that it doesn't do anything but it is provided by the API maybe it should be left undocumented or removed for StageChannels for now - I'm personally not sure if Discord plans on making StageChannels messageable in the future or not. I'd personally just leave it undocumented but ultimately it's up to you and Danny to decide what to do about it.

**Comment by Zephyrkul** [comment]
On looking through the API docs, only text and forum channels are documented to support rate_limit_per_user:
Screenshot from the API docs
As such this PR uses undocumented features of the Discord API.

**Comment by Zephyrkul** [comment]
I found this PR on the discord API docs repo: https://github.com/discord/discord-api-docs/pull/5708
It wouldn't change the documentation for the channel edit endpoint, but it would document that voice and stage channels have this attribute

**Pull request merged**